Add a Python quickstart (#1388)
ngrok/ngrok-docs: docs/getting-started/python.mdx

---
sidebar_label: Python
title: ngrok Python SDK Quickstart
pagination_next: universal-gateway/overview
---

import TabItem from "@theme/TabItem";
import Tabs from "@theme/Tabs";
import ConfigExample from "/src/components/ConfigExample.tsx";
import SecureYourApp from "./_secure-your-app.mdx"

[The ngrok Python SDK](https://ngrok.github.io/ngrok-python/index.html) enables you to quickly and efficiently serve Python applications on the internet without the need to configure low-level network primitives like IPs, certificates, load balancers, or ports.

> This quickstart uses ngrok's Python SDK to put a Python app on the internet and secure it so only you can access it.

Reviewer: reading over this again, I feel like maybe we ought to underscore the fact that the app is running locally

```suggestion
This quickstart uses ngrok's Python SDK to put a Python app running on your local device online and secure it so only you can access it.
```

```python filename="service.py"
from http.server import BaseHTTPRequestHandler, HTTPServer

class MyHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header("Content-type", "text/html")
        self.end_headers()
        html = b"""
        <html>
        <head><title>Test Page</title></head>
        <body><h1>Hello from Python HTTP Server!</h1></body>
        </html>
        """
        self.wfile.write(html)

def run(server_class=HTTPServer, handler_class=MyHandler, port=8080):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f"Serving custom HTML at http://localhost:{port}")
    httpd.serve_forever()

if __name__ == "__main__":
    run()
```

Navigate to the directory where this file is located and run the following command to start the server:

```bash
python service.py
```

## 2. Install the python SDK

Use the following terminal command to install the Python SDK.

```bash
pip install ngrok
```

## 3. Create a .env file

Create a `.env` file and add [your ngrok auth token](https://dashboard.ngrok.com/get-started/your-authtoken) to it.

> This file needs to be in a new directory with the Python script we'll create in the next step that'll use ngrok's SDK to create an endpoint.

Reviewer: one more nitpick - we could trim this down a little so as not to overload the reader with info they don't need yet

```suggestion
This file needs to be in a new directory that will also contain the Python script we'll create in the next step.
```

Create this directory within your existing project directory, navigate into it, and add the `.env` file there.

```bash

> mkdir hello-ngrok && cd hello-ngrok

Reviewer: I'm second-guessing the generic name `hello-ngrok` here. What about `endpoints` or something more descriptive (that you might actually use in the real world)?

import SecureYourApp from "./_secure-your-app.mdx"

[ngrok-python](https://ngrok.github.io/ngrok-python/index.html) is the official Python SDK for ngrok. It enables you to serve Python applications on the internet in a single line of code without configuring low-level network primitives like IPs, certificates, load balancers, or ports.

This quickstart will use the [Agent SDK](/agent-sdks/) for ngrok-python to put a Python app on the internet and secure it so only you can access it.

## What you'll need

1. [An ngrok account](https://dashboard.ngrok.com/signup).
1. Your ngrok authtoken, which you can find [in the dashboard](https://dashboard.ngrok.com/get-started/your-authtoken).
1. [Python installed](https://www.python.org/downloads/) on your machine. You can check this by running `python -v` in your terminal.
1. A domain connected to your ngrok account. We'll refer to this as `$YOUR_DOMAIN` from here on out.

### Get your domain

In the dashboard, visit [the **Domains** section](https://dashboard.ngrok.com/domains) and select **+ New Domain** to get started. ngrok offers free static domains, or you can [bring your own custom domain](/docs/universal-gateway/custom-domains/).

> ## 1. Install `ngrok-python`

Author: ```suggestion
## 1. Install the Python SDK`
```
